### PR TITLE
Replace snappy with bzip2 for compression

### PR DESCRIPTION
**Pull request by ebartels**
@DavW @soundcloud/data-platform-team 

Tested on barn node running against a copy of the barn hierarchy and publishing to test dir on cdh4 hdfs.  Then copying the file from cdh4 to cdh5 to run hadoop-inspect against

```
Sequence File: <org.apache.hadoop.io.LongWritable,org.apache.hadoop.io.BytesWritable>
Compression Type: BLOCK
Compression Codec: org.apache.hadoop.io.compress.BZip2Codec@3a7b2e2
Metadata: size: 0
```

### CODE REVIEW

**Comment by ebartels** [comment]
Artifact is published to http://maven.int.s-cloud.net/#nexus-search;quick~barn and I will do a related system repo update.

File: barn-hdfs/src/main/scala/barn/combiner.scala [hunk 1]
```diff
@@ -90,6 +90,9 @@ trait FileCombiner extends Logging {
   import org.apache.hadoop.io.compress.CompressionCodec
 
   def compressionCodec : CompressionCodec = {
+    import org.apache.hadoop.io.compress.BZip2Codec
+    new BZip2Codec()
```
Comment (DavW) [inline barn-hdfs/src/main/scala/barn/combiner.scala]: I don't get this line. Is this to just make sure it's not filtered out of the dist jar? Where does it actually set the codec to use bz2 instead?

Comment (ebartels) [inline barn-hdfs/src/main/scala/barn/combiner.scala]: @DavW 

compressionCodec is then referenced [here](https://github.com/soundcloud/barn/blob/32498972af656aeb39cbb9155b1e9645cb5fb05d/barn-hdfs/src/main/scala/barn/combiner.scala#L55) and then will result in bzip2 being used.

Comment (ebartels) [inline barn-hdfs/src/main/scala/barn/combiner.scala]: or from what I gather with my limited scala knowledge.

Comment (klorand) [inline barn-hdfs/src/main/scala/barn/combiner.scala]: it sets it in line 55 when it creates the sequence file writer val outputWriter = SequenceFile.createWriter( fs
                         , conf
                         , new HdfsDir(outputFile.getAbsolutePath)
                         , classOf[LW]
                         , classOf[BW]
                         , CompressionType.BLOCK
                         , compressionCodec
                       )

**Pull request merged**